src/modules/musig/main_impl.h

```diff
@@ -12,6 +12,27 @@
 #include "include/secp256k1_musig.h"
 #include "hash.h"
 
+int secp256k1_musig_sort_pubkeys(const secp256k1_context* ctx, secp256k1_xonly_pubkey *pubkeys, size_t n_pubkeys) {
+    size_t i, j;
+    secp256k1_xonly_pubkey key;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(pubkeys != NULL);
+
+    /* insertion sort */
+    /* TODO: replace with more efficient algo. May require scratch space. */
```

Comment:
I am curious as to why the ANSI C qsort function can't be used?

Reply:
is it because it could have "non-linear best case complexity"? would be nice to have that mentioned in the comment.

Reply:
It's because a previous iteration of this PR (hash-by-index) required a more flexible sorting API than qsort (like qsort_r, but that is not available everywhere). I forgot that this could be removed now. Thanks! Pushed fixup commit to replace by qsort.

Reply:
Replaced this again with a heapsort implementation by roconnor implementation, because qsort can be slow on malicious input (as you also mentioned). Also added comment.